src/server/browser.ts

```diff
@@ -89,20 +87,18 @@ export abstract class Browser extends EventEmitter {
     this._downloads.delete(uuid);
   }
 
-  _videoStarted(videoId: string, file: string, pageOrError: Promise<Page | Error>) {
-    const video = new Video(file);
-    this._idToVideo.set(videoId, video);
+  _videoStarted(video: Video, pageOrError: Promise<Page | Error>) {
+    this._idToVideo.set(video._videoId, video);
     pageOrError.then(pageOrError => {
-      // Emit the event in another task to ensure that newPage response is handled before.
       if (pageOrError instanceof Page)
-        makeWaitForNextTask()(() => pageOrError.emit(Page.Events.VideoStarted, video));
```

Comment:
This was necessary for the following code to work:
```js
const page = await context.newPage();
const v = await page.waitForEvent('videostarted');
```
Do we not need this to work any more?

Reply:
Yeah, we don't have an event anymore, so this is not a problem currently.